datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -556,6 +556,9 @@ def send_config_metadata(self):
             'config', self.init_config, section='init_config', whitelist=self.METADATA_DEFAULT_CONFIG_INIT_CONFIG
         )
 
+    def is_metadata_collection_enabled(self):
+        return self.agentConfig.get('enable_metadata_collection', True)
```

Comment:
Since agentConfig is deprecated

```suggestion
        return is_affirmative(datadog_agent.get_config('enable_metadata_collection'))
```